src/spline_agent/commons/proxy.py

```diff
@@ -0,0 +1,80 @@
+#  Copyright 2023 ABSA Group Limited
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from enum import Enum
+from typing import TypeVar, Callable, Any, Tuple, Dict
+
+
+class MemberType(Enum):
```

Comment:
Wouldn't be better to have this Enum class as a standalone class? Just to declare the enums.
It would better follow single responsibility principle.

Reply:
Discussed offline.

General recommendation seems to be:
> **Enums used in class signatures**: If an enum is used in the signature of a class (for example, as the type of a method argument), it's a good idea to define the enum at the top level of the file, above the class definition. This makes it clear that the enum is a separate entity that is used by the class, rather than being part of the class itself.